### ISSUE TITLE
Allow all integers for trust anchor serial numbers

### ISSUE THREAD

**Issue by ctz** [briansmith/webpki#17]
72a3ad9d3c04e8fc226c03577d778c7e0741f366 made parsing serial numbers reject zero and negative serial numbers. There are a few such root certificates in mozilla's and debian's root certs:

```
subject= /C=EU/O=AC Camerfirma SA CIF A82743287/OU=http://www.chambersign.org/CN=Chambers of Commerce Root
serial=00
subject= /C=US/ST=Arizona/L=Scottsdale/O=GoDaddy.com, Inc./CN=Go Daddy Root Certificate Authority - G2
serial=00
subject= /C=US/ST=Arizona/L=Scottsdale/O=Starfield Technologies, Inc./CN=Starfield Root Certificate Authority - G2
serial=00
subject= /C=US/ST=Arizona/L=Scottsdale/O=Starfield Technologies, Inc./CN=Starfield Services Root Certificate Authority - G2
serial=00
subject= /C=EU/O=AC Camerfirma SA CIF A82743287/OU=http://www.chambersign.org/CN=Global Chambersign Root
serial=00
subject= /C=ES/O=Agencia Catalana de Certificacio (NIF Q-0801176-I)/OU=Serveis Publics de Certificacio/OU=Vegeu https://www.catcert.net/verarrel (c)03/OU=Jerarquia Entitats de Certificacio Catalanes/CN=EC-ACC
serial=-11D4C2142BDE21EB579D53FB0C223BFF
subject= /C=US/O=The Go Daddy Group, Inc./OU=Go Daddy Class 2 Certification Authority
serial=00
subject= /C=GR/O=Hellenic Academic and Research Institutions Cert. Authority/CN=Hellenic Academic and Research Institutions RootCA 2011
serial=00
subject= /C=JP/O=SECOM Trust Systems CO.,LTD./OU=Security Communication EV RootCA1
serial=00
subject= /C=JP/O=SECOM Trust Systems CO.,LTD./OU=Security Communication RootCA2
serial=00
subject= /C=JP/O=SECOM Trust.net/OU=Security Communication RootCA1
serial=00
subject= /C=US/O=Starfield Technologies, Inc./OU=Starfield Class 2 Certification Authority
serial=00
```

These are technically incorrect and most are of questionable value, but `aws.amazon.com` roots to `Starfield Class 2 Certification Authority` and `wordpress.com` roots to `Go Daddy Root Certificate Authority - G2`.

**Comment by briansmith** [comment]
Is the known *practical* problem just root certificates? If so, my suggestion here is that we keep disallowing the negative and zero serial numbers in certificates, but allow them in trust anchors. We can do this by generalizing your v1 rust anchor certificate parser to also accept the v3 form of certs.

IMO, it is good to reject invalid serial numbers in (non-root) certificates because it is more likely that revocation systems will get tripped up by invalid serial numbers. Even though I think OCSP and related mechanisms need improvement, we should still try to keep them working as well as we can.

In particular, if we were to accept non-positive serial numbers here, then we'd have to remember to do the same for OCSP and everything else.

**Comment by ctz** [comment]
Yes, I'm only having immediate problems with this for roots. I haven't come across any actually-issued broken leaf/intermediate certificates.

**Comment by briansmith** [comment]
@ctz, please check out https://github.com/briansmith/webpki/commits/more-liberal-trust-anchor-parsing. The last commit on that branch attempts to resolve this issue. Could you add a test case for this based on your real-world certificate? Thanks!

**Comment by ctz** [comment]
Here's such a test case: https://github.com/ctz/webpki/commit/071e323426d8d6e718e92b7aba151da7faaf9144

**Comment by briansmith** [comment]
This was fixed in 8d3309988dc37605708768af60760443c1c2e14d. (If not, please holler.)
